### ISSUE TITLE
dumps twice will cause an error

### ISSUE THREAD

**Issue by Yvictor** [ijl/orjson#101]
I use xxhash will and dumps it twice will let value from `b30d56b4` become `'\x08\x00\x00\x00\x00\x00\x00\x00'`.
error sample code below

``` python
import pytest
import orjson
import xxhash

def test_will_failed():
    content = dict(a=xxhash.xxh32_hexdigest("12345"))
    orjson.dumps(content)
    res = orjson.loads(orjson.dumps(content))
    assert content == res


def test_can_pass():
    content = dict(a=xxhash.xxh32_hexdigest("12345"))
    res = orjson.loads(orjson.dumps((content)))
    assert res == content
```

**Comment by ijl** [comment]
This looks like a problem with how xxhash is constructing the `str`.

```
>>> import orjson, xxhash
>>> hexdigest = xxhash.xxh32_hexdigest("12345")
>>> orjson.dumps(hexdigest)
b'"b30d56b4"'
>>> orjson.dumps(hexdigest)
b'"\\b\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000"'
>>> hexdigest_copied = xxhash.xxh32_hexdigest("12345").encode("utf-8").decode("utf-8")
>>> orjson.dumps(hexdigest_copied)
b'"b30d56b4"'
>>> orjson.dumps(hexdigest_copied)
b'"b30d56b4"'
```

Notice that if I recreate the `str` using CPython it's fine. You could mitigate by wrapping the hash function like that.

**Comment by ijl** [comment]
To be fixed in 3.1.1.
